outreach/write.go

```diff
@@ -18,25 +18,36 @@ var JSONAPIContentTypeHeader = common.Header{ //nolint:gochecknoglobals
 }
 
 func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
-	var write common.WriteMethod
+	var (
+		write common.WriteMethod
+		data  map[string]any
+		err   error
+	)
 
 	url, err := c.getApiURL(config.ObjectName)
 	if err != nil {
 		return nil, err
 	}
 
+	// prepares the updating data request.
 	if len(config.RecordId) > 0 {
 		url.AddPath(config.RecordId)
 
 		write = c.Client.Patch
+
+		data, err = parseData(config.RecordData, config.ObjectName, config.RecordId)
```

Comment:
Thanks, I also did not like slice of ids... What do you say if I just change the signature of this function to be `parseData(config)`. Since am using all the writeParam details anyway.  And this will remove the slice of ids concern.

Reply:
You know that's even better.